lib/idx/authenticate.ts

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable complexity */
+import { AuthTransaction } from '../tx';
+import { OktaAuth, AuthorizeOptions, IdxTransactionMeta, RemediationValues } from '../types';
+import { interact } from './interact';
+import { introspect } from './introspect';
+import { remediate } from './remediate';
+
+export async function authenticate(authClient: OktaAuth, options: AuthorizeOptions): Promise<AuthTransaction> {
+  let { state, scopes, interactionHandle } = options;
+
+  if (interactionHandle) {
+    // resume existing transaction, not supported yet
```

Comment:
should we assume this method is the entry point of the idx flow, and the `interactionHandle` should always be null?

Reply:
I think for re-entrant flows (like social IDP) we will call `authenticate` again using a saved interaction handle.